src/Orchard.Web/Modules/Orchard.CustomForms/Controllers/ItemController.cs

```diff
@@ -198,6 +198,13 @@ public class ItemController : Controller, IUpdateModel {
 
             contentItem.As<ICommonPart>().Container = customForm.ContentItem;
 
+            // save the submitted form
```

Comment:
Is it worth creating a new "submitting" event instead? Can anything be done between them of value (The workflow is triggered so something could be done).

Reply:
As explained here https://github.com/OrchardCMS/Orchard/pull/8795#discussion_r1596347011 , I tested how to make the workflow work and the only way to have the actual values of the fields was to publish the content item and reading it again.
For this reason, I moved the logic I tested from the RazorExecute activity to the controller to make it work as expected without publishing the content explicitly inside the external code.

For reference, this is how our worflow looks like:
![image](https://github.com/OrchardCMS/Orchard/assets/83577153/80bdff01-ac1d-49e5-bdbd-9adfce5473b6)

The "SelezionaDestinatari.cshtml" razor reads a TaxonomyField inside the content item created by the form submit to make some operations. If content isn't published (and re-read), the TaxonomyField is empty.